src/__internal__/focus-trap/focus-trap.component.js

```diff
@@ -157,18 +165,53 @@ const FocusTrap = ({
     ) {
       // the trap breaks if it tries to refocus a disabled element
       setElementFocus(currentFocusedElement);
+    } else if (wrapperRef?.current?.hasAttribute("tabindex")) {
+      setElementFocus(wrapperRef.current);
     } else if (firstElement) {
       setElementFocus(firstElement);
     }
-  }, [currentFocusedElement, firstElement]);
+  }, [currentFocusedElement, firstElement, wrapperRef]);
 
   useEffect(() => {
     if (triggerRefocusFlag) {
       refocusTrap();
     }
   }, [triggerRefocusFlag, refocusTrap]);
 
-  return <div ref={trapRef}>{children}</div>;
+  const [tabIndex, setTabIndex] = useState(0);
+
+  useEffect(() => {
+    // ideally would set to -1, but that fails Cypress tests due to Cypress bug.
```

Comment:
I think this could just a one line comment something like below

```
Issue in cypress prevents setting tabIndex to -1 (link to issue if there is one), instead tabIndex is removed onBlur and reset on close